design/ceph/object/store.md

```diff
@@ -323,31 +323,113 @@ spec:
+The below example illustrates a commonly-anticipated configuration, and one that Rook can recommend
+to users. The user is using an internal, wildcard-enabled K8s service as the advertised endpoint for
+OBCs. The user also has externally-available ingress that the RGW needs to accept connections from,
+which serves S3 applications outside the Kubernetes cluster. Both endpoints are wildcard-enabled,
+but the internal service is preferable to the external ingress to prevent the ingress router from
+being a cluster-wide S3 bottleneck for Kubernetes applications.
 
 ```yaml
 spec:
   hosting:
+    advertiseEndpoint:
+      dnsName: my-internal-wildcard-service.mydomain.com
+      port: 8443
+      useTLS: true
     dnsNames:
-    - "my-ingress.mydomain.com"
+      - my-external-ingress.mydomain.com
 ```
 
-A list of hostnames to use for accessing the bucket directly like a subdomain in the endpoint. For example if the ingress service endpoint `my-ingress.mydomain.com` added and the object store contains a bucket named `sample`, then the s3 [virtual host style](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html) would be `http://sample.my-ingress.mydomain.com`. More details about the feature can be found in [Ceph Documentation](https://docs.ceph.com/en/latest/radosgw/s3/commons/#bucket-and-host-name).
-
-When this feature is enabled the endpoint in OBCs and COSI Bucket Access can be clubbed with bucket name and host name. For OBC  the `BUCKET_NAME` and the `BUCKET_HOST` from the config map combine to `http://$BUCKETNAME.$BUCKETHOST`. For COSI Bucket Access the `bucketName` and the `endpoint` in the `BucketInfo` to `http://bucketName.endpoint`.
+Each `rgw_dns_name` entry must be a valid RFC-1123 hostname, and Rook Operator will perform input validation using k8s apimachinery `IsDNS1123Subdomain()`.
+
+When `rgw_dns_name` is changed for an RGW cluster, all RGWs need to be restarted. To enforce this in Rook, we can apply the `--rgw-dns-name` flag, which will restart RGWs with no user action needed.
+
+This is different from CephObjectZone `customEndpoints` which is used for configuring the object store to replicate and sync data amongst Multisite.
+
+When endpoints are configured in `rgw_dns_names`, the RGW will reject any incoming connections
+intended for endpoints not in the list. Therefore, all endpoints that might be used must be added.
+
+For convenience, and to ensure other CephObjectStore configurations are not rendered unusable when
+users add additional endpoints to `dnsNames`, the following should be added to the `rgw_dns_names`
+list automatically by Rook:
+- the `advertiseEndpoint.dnsName`
```

Comment:
If the advertise endpoint is also in the customEndpoints list, will we remove the duplicate? Or do we require them not to set the advertise endpoint to something already in the customEndpoints?

Reply:
Looks like I accidentally deleted the duplicate entry design note. I'll add it back in.